File: SocketJack.NewtonsoftJson/Int32NotSupportedException.vb
Public Class Int32NotSupportedException
    Inherits Exception

    Public Overrides ReadOnly Property Message As String = "Change the Class's Property listed in Int32NotSupportedException.AffectedProperty to Int64."

    ''' <summary>
    ''' This property should be type of Int64.
    ''' </summary>
    ''' <returns></returns>
    Public ReadOnly Property AffectedProperty As String
        Get
            Return _AffectedProperty
        End Get
    End Property
    Private _AffectedProperty As String
    Public Sub New(AffectedProperty As String)
        _AffectedProperty = AffectedProperty
    End Sub
End Class

File: SocketJack/Serialization/Json/Int32NotSupportedException.cs
using SocketJack;
using SocketJack.Serialization;
using SocketJack.Serialization.Json;
using System;

namespace SocketJack.Serialization.Json {
    public class Int32NotSupportedException : Exception {

        public override string Message { get; } = "Change the Class's Property listed in Int32NotSupportedException.AffectedProperty to Int64.";

        /// <summary>
    /// This property should be type of Int64.
    /// </summary>
    /// <returns></returns>
        public string AffectedProperty {
            get {
                return _AffectedProperty;
            }
        }
        private string _AffectedProperty;
        public Int32NotSupportedException(string AffectedProperty) {
            _AffectedProperty = AffectedProperty;
        }
    }
}